Samples/WinMLSamplesGallery/TransformAsync/External/CSampleQueue.cpp

```diff
@@ -0,0 +1,289 @@
+#include "pch.h"
+#include "CSampleQueue.h"
+
+class CSampleQueue::CNode
+{
+public:
+    IMFSample*  pSample;
+    CNode*      pNext;
+
+    CNode(void)
+    {
+        pSample = NULL;
+        pNext   = NULL;
+    }
+    ~CNode(void)
+    {
+        SAFE_RELEASE(pSample);
+        pNext   = NULL;
+    }
+};
+
+HRESULT CSampleQueue::Create(
+    CSampleQueue**   ppQueue)
+{
+    HRESULT         hr          = S_OK;
+    CSampleQueue*   pNewQueue   = NULL;
+
+    do
+    {
+        if(ppQueue == NULL)
+        {
+            hr = E_POINTER;
+            break;
+        }
+
+        pNewQueue = new CSampleQueue();
+        if(pNewQueue == NULL)
+        {
+            hr = E_OUTOFMEMORY;
+            break;
+        }
+
+        (*ppQueue) = pNewQueue;
+        (*ppQueue)->AddRef();
+    }while(false);
+
+    SAFE_RELEASE(pNewQueue);
+
+    return hr;
+}
+
+ULONG CSampleQueue::AddRef(void)
```

Comment:
[nit] Appears this is written as C code, even though it's a .cpp extension. Propose we `(void)` -> `()`.